### PR TITLE
{bio,lib}[foss/2021a] mapDamage v2.2.1, inline v0.3.19, seqtk v1.3, ... w/ Python 3.9.5

### PR DESCRIPTION
**Pull request by jfgrimm**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/m/mapDamage/mapDamage-2.2.1-foss-2021a-R-4.1.0.eb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    ('Pysam', '0.16.0.1'),
+    ('R', '4.1.0'),
+    ('inline', '0.3.19', versionsuffix),
+    ('RcppGSL', '0.3.10', versionsuffix),
+]
+
+use_pip = True
+sanity_pip_check = True
+download_dep_fail = True
+
```
Comment (branfosj) [inline easybuild/easyconfigs/m/mapDamage/mapDamage-2.2.1-foss-2021a-R-4.1.0.eb:32]: I tried adding `sanity_check_commands = ['%(name)s --help']` here, but that gives me a failure of:
```
== 2021-12-18 11:01:41,712 easyblock.py:3385 WARNING Sanity check: sanity check command mapDamage --help exited with code 1 (output: Seqtk executable not accessible; mapDamage has not
been intalled properly or current user does not
sufficient permissions. Expected executable at
   '/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/mapDamage/2.2.1-foss-2021a-R-4.1.0/lib/python3.9/site-packages/mapdamage/seqtk/seqtk'
)
```
So there is some hardcoding in there on where `mapDamage` expects `seqtk` to be.

**Review by branfosj — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/m/mapDamage/mapDamage-2.2.1-foss-2021a-R-4.1.0.eb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+sources = ['%(version)s.tar.gz']
+checksums = ['f86cd2b1fac3e6a1d6ccc97d37ef6cf3b5be07747c3d3894472a8d1da9eb595c']
+
+dependencies = [
+    ('Python', '3.9.5'),
+    ('seqtk', '1.3'),
+    ('Pysam', '0.16.0.1'),
+    ('R', '4.1.0'),
+    ('inline', '0.3.19', versionsuffix),
+    ('RcppGSL', '0.3.10', versionsuffix),
```
[the remark this answers is not in the document]
Comment (jfgrimm) [reply easybuild/easyconfigs/m/mapDamage/mapDamage-2.2.1-foss-2021a-R-4.1.0.eb:26]: doubt it

File: easybuild/easyconfigs/m/mapDamage/mapDamage-2.2.1_use-external-seqtk.patch [hunk 3]
```diff
@@ -0,0 +1,74 @@
+Author: Jasper Grimm (UoY)
+Allow mapDamage to use a seqtk from $PATH, instead of looking in its install dir
```
Comment (boegel) [inline easybuild/easyconfigs/m/mapDamage/mapDamage-2.2.1_use-external-seqtk.patch:2]: @jfgrimm To what extend did you test whether using an external `seqtk` with `mapDamage` is working correctly?

We got a problem report from a researcher trying to use the `mapDamage` module that was installed with the easyconfig file from #14725, which fails with:

```
Seqtk executable not accessible; mapDamage has not
been intalled properly or current user does not
sufficient permissions. Expected executable at
   '/software/mapDamage/2.2.1-foss-2021b/lib/python3.9/site-packages/mapdamage/seqtk/seqtk'
```

This patch would fix that, clearly, but apparently the `seqtk` that's included with `mapDamage` is a heavily patched/trimmed down version, compared to the original (`diff` produces a 2000-line patch on `seqtk.c`).

Comment (jfgrimm) [reply easybuild/easyconfigs/m/mapDamage/mapDamage-2.2.1_use-external-seqtk.patch:2]: as far as I understand, the `seqtk` they ship is simply a stripped down version, but a full `seqtk` should still be compatible. FWIW the conda recipe uses a patch to similar effect, and also uses an fullexternal `seqtk`, and that's listed as one of the recommended installation methods on mapDamage's homepage

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3119.skitty.os - Linux UNKNOWN UNKNOWN, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/d7872a54a8388277486d95a85cb6b7c3 for a full test report.

File: easybuild/easyconfigs/m/mapDamage/mapDamage-2.2.1_use-external-seqtk.patch (hunk 3 above)
Comment (boegel) [reply easybuild/easyconfigs/m/mapDamage/mapDamage-2.2.1_use-external-seqtk.patch:2]: Excellent, thanks for clarifying!

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3523.doduo.os - Linux UNKNOWN UNKNOWN, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/c56b3c74c693a5df5bce6375e13de584 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @jfgrimm!

**Pull request merged**